misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -89,29 +91,112 @@
         String foundTypeLeft = matcher.group(1);
         String foundTypeRight = matcher.group(2);
 
+        boolean foundErrorLeft = false;
+        boolean foundErrorRight = false;
+
+        // If the project has multiple modules
+        if (foundTypeLeft.contains(":")) {
+            foundTypeLeft = foundTypeLeft.split(":")[foundTypeLeft.split(":").length - 1];
```

Comment:
I think this truncation in symbol name causes invalid symbol matching. For instance; what if the assigning record is from a different module and yet you have another record with the same name in current module.`rasika/proj.mod1:0.1.0:Grades` and `abc/proj.othermodule:0.1.0:Grades` will be taken as equal(which should not be the case).  You need to match moduleIds and symbolNames both.

Even with the new api support from the diagnostics-api mentioned in next comment, I think we don't really need this symbol names matching.

Reply:
Removed this from the PR and will move this issue to a separate PR.